### PR TITLE
feat: add copy and download buttons to 2fa recovery modal

### PR DESCRIPTION
**Pull request by mc-jones**
This PR adds functionality allowing users to download and copy their 2FA recovery codes during the 2FA setup flow, or when they are requested via the settings page. This implementation is from @lidimayra's #8930 PR, now that the component have been moved to palette v3 with #9048 .

Screen Shot 2021-12-14 at 2 05 28 PM

## Rational
As more partners are enrolled into 2FA, we are seeing an increase in account lockouts as users lose access to their 2FA authentication method(s). Unlocking an account requires the Trust and Safety and Platform teams to validate the ID of the requester and manually lookup and provide a code.  This functionality specifically aims to increase backup code utilization through improved UX. 

[PLATFORM-3280]

[PLATFORM-3280]: https://artsyproduct.atlassian.net/browse/PLATFORM-3280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesActions.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export const SettingsEditSettingsTwoFactorBackupCodesActions: React.FC<SettingsEditSettingsTwoFactorBackupCodesActionsProps> = props => {
+  const { backupSecondFactors } = props
+
+  const [supportsClipboard, setSupportsClipboard] = useState(false)
+
+  useEffect(() => {
+    // Only render the copy button if browser supports the Clipboard API
+    // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
+    if ("clipboard" in navigator) setSupportsClipboard(true)
```
Comment (damassi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesActions.tsx:17]: This is a good check, but I _think_ all browsers now support this. In any case, this makes things a bit safer.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesDialog.tsx [hunk 2]
```diff
@@ -26,19 +27,26 @@ const SettingsEditSettingsTwoFactorBackupCodesDialog: FC<SettingsEditSettingsTwo
...
+                  </Text>
+                </Column>
+              )
+            })}
+          </GridColumns>
+          <SettingsEditSettingsTwoFactorBackupCodesActions
+            backupSecondFactors={me.backupSecondFactors!.map(factor =>
+              factor!.code!.toString()
+            )}
+          ></SettingsEditSettingsTwoFactorBackupCodesActions>
```
Comment (damassi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesDialog.tsx:48]: Tiny nit -- this `SettingsEditSettingsTwoFactorBackupCodesActions` can be self-closing

**Review by damassi — APPROVED** [review]
Looks good! A couple minor comments but nothing blocking 👍

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesDialog.tsx (hunk 2 above)
Comment (mc-jones) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesDialog.tsx]: Ah. Good catch. Fixed.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesActions.tsx (hunk 1 above)
Comment (mc-jones) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesActions.tsx:17]: For sure. Fortunately, all major browsers do support it now, but Edge and Safari support are more recent than I would have thought (2020).

**Review by lidimayra — APPROVED** [review]
Lots of learnings on this, thanks a lot, @mc-jones 🚀

**Pull request merged**